src/components/filters/FilterModal.jsx

```diff
@@ -1,44 +1,82 @@
+import { useFilterNotes, useNotes } from "contexts";
 import "./filtermodal.css";
 export function FilterModal({ setShowFilterModal }) {
-  const labels = ["label1", "label2", "label3", "label4"];
+  const { tags } = useNotes();
+  const { stateFilter, dispatchFilter } = useFilterNotes();
+  const { priority, sortByDate, tags: filterTags } = stateFilter;
   return (
     <>
       <div className="d-flex modal-container">
         <div className="modal children-stacked">
           <>
-            <div className="modal-header d-flex">
-              <h5>Filter by:</h5>
-            </div>
+            <h5 className="text-center">Filter by:</h5>
             <div className="modal-body">
               <h6>Tags:</h6>
               <ul className="d-flex grid-gap filter-tags">
-                {labels &&
-                  labels?.map(item => {
-                    return (
-                      <li className="d-flex">
-                        <span className="">
-                          <label>
-                            <input type="checkbox" /> &nbsp;
-                            {item}
-                          </label>
-                        </span>
-                      </li>
-                    );
-                  })}
+                {tags?.map((item, idx) => {
+                  return (
+                    <li className="d-flex" key={"filter-tag" + idx}>
+                      <span className="">
+                        <label>
+                          <input
+                            type="checkbox"
+                            id={item}
+                            name={item}
+                            onChange={() => {
+                              dispatchFilter({
+                                type: "SET_TAG",
+                                payload: item,
+                              });
+                            }}
+                            checked={
+                              filterTags.find(t => t === item) ? true : false
```

Comment:
Might wanna name them as tags or something more meaningful. This doesn't give a clear understanding.